### PR TITLE
feat(CloudFormation): Include EnableTerminationProtection

### PR DESCRIPTION
**Pull request by jfagoagas**
### Description

Include the `EnableTerminationProtection` in the CFN API `Create_Stack` and `Describe_Stacks`.

### CODE REVIEW

**Comment by jfagoagas** [comment]
I think the errors thrown [here](https://github.com/spulec/moto/actions/runs/3417205170/jobs/5688101883) are not related with this PR:
```
Running MyPy...
mypy --install-types --non-interactive
Collecting types-PyYAML
  Downloading types_PyYAML-6.0.12.1-py3-none-any.whl (14 kB)
Collecting types-python-dateutil
  Downloading types_python_dateutil-2.8.19.2-py3-none-any.whl (8.1 kB)
Collecting types-pytz
  Downloading types_pytz-2022.6.0.1-py3-none-any.whl (4.7 kB)
Collecting types-requests
  Downloading types_requests-2.28.11.2-py3-none-any.whl (14 kB)
Collecting types-urllib3<1.27
  Downloading types_urllib3-1.26.[25](https://github.com/spulec/moto/actions/runs/3417205170/jobs/5688101883#step:8:26).1-py3-none-any.whl (14 kB)
Installing collected packages: types-urllib3, types-PyYAML, types-pytz, types-python-dateutil, types-requests
Successfully installed types-PyYAML-6.0.12.1 types-python-dateutil-2.8.19.2 types-pytz-2022.6.0.1 types-requests-2.28.11.2 types-urllib3-1.[26](https://github.com/spulec/moto/actions/runs/3417205170/jobs/5688101883#step:8:27).25.1
moto/apigateway/models.py:70:1: error: Metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases  [misc]
Installing missing stub packages:
/opt/hostedtoolcache/Python/3.7.15/x64/bin/python -m pip install types-PyYAML types-python-dateutil types-pytz types-requests


Found 16 errors in 1 file (checked 139 source files)
moto/apigateway/models.py:70:1: note: Error code "misc" not covered by "type: ignore" comment
moto/apigateway/models.py:105:1: error: Metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases  [misc]
moto/apigateway/models.py:105:1: note: Error code "misc" not covered by "type: ignore" comment
moto/apigateway/models.py:1[28](https://github.com/spulec/moto/actions/runs/3417205170/jobs/5688101883#step:8:29):1: error: Metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases  [misc]
moto/apigateway/models.py:128:1: note: Error code "misc" not covered by "type: ignore" comment
moto/apigateway/models.py:183:1: error: Metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases  [misc]
moto/apigateway/models.py:183:1: note: Error code "misc" not covered by "type: ignore" comment
moto/apigateway/models.py:196:1: error: Metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases  [misc]
moto/apigateway/models.py:196:1: note: Error code "misc" not covered by "type: ignore" comment
moto/apigateway/models.py:460:1: error: Metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases  [misc]
moto/apigateway/models.py:460:1: note: Error code "misc" not covered by "type: ignore" comment
moto/apigateway/models.py:514:1: error: Metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases  [misc]
moto/apigateway/models.py:514:1: note: Error code "misc" not covered by "type: ignore" comment
moto/apigateway/models.py:667:1: error: Metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases  [misc]
moto/apigateway/models.py:667:1: note: Error code "misc" not covered by "type: ignore" comment
moto/apigateway/models.py:711:1: error: Metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases  [misc]
moto/apigateway/models.py:711:1: note: Error code "misc" not covered by "type: ignore" comment
moto/apigateway/models.py:753:1: error: Metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases  [misc]
moto/apigateway/models.py:753:1: note: Error code "misc" not covered by "type: ignore" comment
moto/apigateway/models.py:805:1: error: Metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases  [misc]
moto/apigateway/models.py:805:1: note: Error code "misc" not covered by "type: ignore" comment
moto/apigateway/models.py:814:1: error: Metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases  [misc]
moto/apigateway/models.py:814:1: note: Error code "misc" not covered by "type: ignore" comment
moto/apigateway/models.py:1208:1: error: Metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases  [misc]
moto/apigateway/models.py:1208:1: note: Error code "misc" not covered by "type: ignore" comment
moto/apigateway/models.py:1244:1: error: Metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases  [misc]
moto/apigateway/models.py:1244:1: note: Error code "misc" not covered by "type: ignore" comment
moto/apigateway/models.py:1257:1: error: Metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases  [misc]
moto/apigateway/models.py:1257:1: note: Error code "misc" not covered by "type: ignore" comment
moto/apigateway/models.py:1[29](https://github.com/spulec/moto/actions/runs/3417205170/jobs/5688101883#step:8:30)0:1: error: Metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases  [misc]
moto/apigateway/models.py:1290:1: note: Error code "misc" not covered by "type: ignore" comment
make: *** [Makefile:[31](https://github.com/spulec/moto/actions/runs/3417205170/jobs/5688101883#step:8:32): lint] Error 1
Error: Process completed with exit code 2.
```

**Comment by bblommers** [comment]
> I think the errors thrown [here](https://github.com/spulec/moto/actions/runs/3417205170/jobs/5688101883) are not related with this PR:

> ```

That's indeed unrelated @jfagoagas - master is currently broken due to a new MyPy release. Once https://github.com/spulec/moto/pull/5646 is merged, I'll have another look at this PR.

**Comment by jfagoagas** [comment]
> > I think the errors thrown [here](https://github.com/spulec/moto/actions/runs/3417205170/jobs/5688101883) are not related with this PR:
> 
> 
> 
> > ```
> 
> 
> 
> That's indeed unrelated @jfagoagas - master is currently broken due to a new MyPy release. Once https://github.com/spulec/moto/pull/5646 is merged, I'll have another look at this PR.

Thanks!!

File: tests/test_cloudformation/test_cloudformation_stack_crud_boto3.py [hunk 1]
```diff
@@ -312,9 +312,24 @@
 def test_create_stack():
     cf_conn = boto3.client("cloudformation", region_name="us-east-1")
     cf_conn.create_stack(StackName="test_stack", TemplateBody=dummy_template_json)
+    stack = cf_conn.describe_stacks()["Stacks"][0]
+    stack.should.have.key("StackName").equal("test_stack")
```
Comment (bblommers) [inline tests/test_cloudformation/test_cloudformation_stack_crud_boto3.py:316]: Can we add an assertion to the existing test case, `stack.should.have.key("EnableTerminationProtection").equal(False)`?
Just to ensure that the logic is correct when the parameter is not provided

**Review by bblommers — CHANGES_REQUESTED** [review]
The mypy-issues are now fixed, so if you merge in the latest master it should work again

File: tests/test_cloudformation/test_cloudformation_stack_crud_boto3.py (hunk 1 above)
Comment (jfagoagas) [reply tests/test_cloudformation/test_cloudformation_stack_crud_boto3.py:316]: Sure! Included now.

**Review by bblommers — APPROVED** [review]
LGTM! Thanks for the PR @jfagoagas

**Pull request merged**